modules/openapi-generator/src/main/resources/typescript-nestjs/api.module.mustache

```diff
@@ -0,0 +1,40 @@
+import { DynamicModule, HttpService, HttpModule, Module, Global } from '@nestjs/common';
+import { Configuration } from './configuration';
+import { BASE_PATH } from './variables';
+
+{{#apiInfo}}
+{{#apis}}
+import { {{classname}} } from './{{importPath}}';
+{{/apis}}
+{{/apiInfo}}
+
+@Global
+@Module({
+  imports:      [ HttpModule ],
```

Comment:
the naming conflict can be avoided with renamed imports, see also https://github.com/OpenAPITools/openapi-generator/issues/4101

Reply:
Imports can be renamed and it's not the problem -- what I'm worried is that the class name becomes the service/module's token in Nest's dependency injection system. I don't know precisely what will happen if you register two Nest modules/providers under the same name, but it most likely won't work. Example: the generated API clients inject the `Configuration` provider and the `BASE_PATH` value, which will be ambiguous since they are registered globally for all imported `ApiModule`s.

I see it actually [was fixed](https://github.com/nestjs/nest/pull/3363) for module classes, but:
* the fix was made in a relatively recent Nest version (6.10),
* the fix handles module classes, so provider classes (like `Configuration` here) may still not work
* IMHO it's just a bad practice, since it's very likely to import multiple different ApiModules in one app, so it decreases code readability and forces you to manually rename imports every time.
* note that the `BASE_PATH` injected config I mentioned in my other comment is a manually passed DI token, so it will always be ambiguous.

Reply:
true, so this is a bit different from the angular case.

Reply:
@quezak 
1. Because that was a consistent name, the folder structure and the import as give a precise name. Although the cli could have an optional name to specify what you would like. This PR did not seem like the time to work out other enhancements.
2. Global so that a consumer does not have to add the granularity which you are suggesting. The use I am using this fork .jar to codege for is one app initailization in a AWS lambda function. I do not need the granularity you suggest.

Reply:
@quezak class names do not become the Nestjs DI token because they are dynamic modules due to .forRoot(), if they did not have `.forRoot()` it would be a static module

`Configuration` and `BASE_PATH` as seen in the example readme are set by the factory pattern and not a problem as you are possibly stating.